lib/Net/Kubernetes/Role/APIAccess.pm

```diff
@@ -121,30 +130,58 @@ around BUILDARGS => sub {
 
 sub _create_default_base_path {
 	my($self) = @_;
-	return '/api/'.$self->api_version;
+	
+	return '/api/' . $self->api_version;
 }
 
 sub path {
 	my($self) = @_;
 	return $self->url.$self->base_path;
 }
 
+sub _build_server_version {
+	my $self = shift;
+
+	my($version_info) = $self->send_request($self->create_request(GET => $self->url . '/version'));
```

Comment:
Sort of, not really. One structural problem this module has is the fact that most objects are also api clients. That leads to having to pass on all of the api related attributes upon creation of another object. When the server_version is passed on, an object will not ask the server for its version again.

I made sure to pass that variable on in the ResourceFactory role, but I see that I've missed it in the HasPods and HasReplicaSets roles.

If I ever get around to it, I think a major point release would be in order to restructure this API model, making the client an actual object and passing the client around.